crates/turborepo-auth/src/lib.rs

```diff
@@ -31,17 +38,143 @@ impl Token {
     pub fn existing(token: String) -> Self {
         Self::Existing(token)
     }
-    /// Checks if the token is valid. We do a few checks:
+    /// Checks if the token is still valid. The checks ran are:
+    /// 1. If the token is active.
+    /// 2. If the token has access to the cache.
+    ///     - If the token is forbidden from accessing the cache, we consider it
+    ///       invalid.
+    /// 3. We are able to fetch the user associated with the token.
+    ///
+    /// ## Arguments
+    /// * `client` - The client to use for API calls.
+    /// * `valid_message_fn` - An optional callback that gets called if the
+    ///   token is valid. It will be passed the user's email.
+    // TODO(voz): This should do a `get_user` or `get_teams` instead of the caller
```

Comment:
Does this TODO still apply now that we're calling `self.user`?

Reply:
Nope, missed this before merging, so I'll clean it up somewhere in this stack.